### PR TITLE
🧿 Ajna flow & order bugfixes

### PR DESCRIPTION
**Pull request by piotrkonowrocki**
# 🧿 Ajna flow & order bugfixes

Fixed some major and minor bugs related to flow component and order information.
  
## Changes 👷‍♀️

- Fixed issue where order information was shown with delay, only after initial data was loaded,
- fixed issue where primary button was disabled and put into loading state with a debounce,
- fixed issue with existing position where there was no dpm address provided for simulation,
- fixed issue with refreshing simulation data after rpm is created.
  
## How to test 🧪
See if issue mentioned above no longer occur.

### CODE REVIEW

File: blockchain/config.ts [hunk 1]
```diff
@@ -547,6 +547,7 @@ const hardhat: NetworkConfig = {
   cacheApi: 'https://oazo-bcache-mainnet-staging.new.oasis.app/api/v1',
 }
 
+export const nullWalletAddress = '0x0000000000000000000000000000000000000000'
```
Comment (piekczyk) [inline blockchain/config.ts:550]: ethers.constants.AddressZero

File: features/ajna/borrow/useAjnaTxHandler.ts [hunk 2]
```diff
@@ -32,14 +32,29 @@ export function useAjnaTxHandler(): AjnaTxHandler {
 
   const [txData, setTxData] = useState<AjnaTxData>()
   const [cancelablePromise, setCancelablePromise] = useState<CancelablePromise<AjnaActionData>>()
-  const { dpmAddress } = state
 
+  const { depositAmount, generateAmount, paybackAmount, withdrawAmount, dpmAddress } = state
+
+  useEffect(() => {
+    if (txHelpers) {
```
Comment (piekczyk) [inline features/ajna/borrow/useAjnaTxHandler.ts:39]: I know that I initially introduced it but we don't need this condition here

File: features/ajna/borrow/useAjnaTxHandler.ts [hunk 3]
```diff
@@ -32,14 +32,29 @@ export function useAjnaTxHandler(): AjnaTxHandler {
...
+    paybackAmount?.toString(),
+    withdrawAmount?.toString(),
+  ])
   useDebouncedEffect(
     () => {
-      if (txHelpers && context && dpmAddress) {
-        cancelablePromise?.cancel()
-        setIsLoadingSimulation(true)
-
+      if (txHelpers && context) {
```
Comment (piekczyk) [inline features/ajna/borrow/useAjnaTxHandler.ts:57]: please remove `txHelpers`  from this condition and add `context.rpcProvider` to dependency array

**Review by piekczyk — CHANGES_REQUESTED** [review]

File: blockchain/config.ts (hunk 1 above)
Comment (piotrkonowrocki) [reply blockchain/config.ts]: I did not know that, fixed.

File: features/ajna/borrow/useAjnaTxHandler.ts (hunk 2 above)
Comment (piotrkonowrocki) [reply features/ajna/borrow/useAjnaTxHandler.ts]: Fixed.

File: features/ajna/borrow/useAjnaTxHandler.ts (hunk 3 above)
Comment (piotrkonowrocki) [reply features/ajna/borrow/useAjnaTxHandler.ts]: Fixed.

**Review by piekczyk — DISMISSED** [review]

**Review by piekczyk — APPROVED** [review]

**Pull request merged**